bindings/go/src/ucx/endpoint_params.go

```diff
@@ -41,6 +41,9 @@ func ucxgo_completeGoErrorHandler(user_data unsafe.Pointer, ep C.ucp_ep_h, statu
 func (p *UcpEpParams) SetUcpAddress(a *UcpAddress) *UcpEpParams {
 	p.params.field_mask |= C.UCP_EP_PARAM_FIELD_REMOTE_ADDRESS
 	p.params.address = a.Address
+	if p.params.field_mask == C.UCP_EP_PARAM_FIELD_LOCAL_ADDR {
```

Comment:
A bit strange, why needed?

Reply:
There was go tests error which can be run successfully locally on my servers. I suspected it was due to GOCACHE not updated, hence added this to update GOCACHE to verify my thought. I will remove this strange change in the final diff.

From go help cache:
the build cache does not detect changes to C libraries imported with cgo. If you have made changes to the C libraries on your system, you will need to clean the cache explicitly or else use the -a build flag (see 'go help build') to force rebuilding of packages that depend on the updated C libraries. 

In the fix I added variable in GO imported ucp_ep_params_t

Reply:
@ybedbug does it mean that we have to fix our GO building to always clean GOCACHE?

Reply:
Yes. I think so.

Reply:
@dmitrygx Do you know who I can contact to talk about this GOCACHE issue? So that later the change can pass CI tests without go files' change?

Reply:
@ybedbug, @petro-rudenko is the right person to check GO-related things in UCX.

Reply:
Hi, did you see go error in CI? Could you please give a link to the failed job

Reply:
@petro-rudenko 
I got CI errors(2 go errors) https://github.com/openucx/ucx/runs/4126432003. 
I reproduced it on hpchead and after I changed the GOCACHE to new directory had null, it passed on hpchead.
Also I found after I made some change in the go code(no need in fact), the cache would be updated hence tests passed. That is why the latest CI all passed.

Reply:
It works for me without these changes in go. All tests passed. Can you please remove it and recommit.

Reply:
@petro-rudenko I recommitted without go side change. The two go tests failed similar to previous ones. https://github.com/openucx/ucx/pull/7623/checks?check_run_id=4160106002
IMO, if we can clean GOCACHE on the build server, it should pass.

Reply:
I see an error:
```
[1636508072.993971] [swx-rdmz-ucx-gpu-01:5415 :1]          ucp_ep.c:747  UCX  ERROR sa_data_v1 contains unsupported address mode 73
```

Reply:
@petro-rudenko this error was due to GOCACHE was old. The go file such as endpoint_params.go it imported C.ucp_ep_params_t, this ucp_ep_params_t was changed in C in my diff. I reproduced this on one server and error was gone by clean the GOCACHE.

From go help cache:
the build cache does not detect changes to C libraries imported with cgo. If you have made changes to the C libraries on your system, you will need to clean the cache explicitly or else use the -a build flag (see 'go help build') to force rebuilding of packages that depend on the updated C libraries.

Reply:
@petro-rudenko Not all servers can reproduce the two errors, it depends on the GOCACHE on the build servers. I just happened to find one server that can reproduce this. But I think our CI should not depend on GOCACHE, it should clean it to avoid further such issues i.e. imported C change not detected by GOCACHE.

Reply:
thanks to Peter's fix. Go test errors were fixed.